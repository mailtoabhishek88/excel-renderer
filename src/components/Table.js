import React from 'react';
import ReactDOM from 'react-dom';
import {ExcelRenderer} from 'react-excel-renderer';
import Row from './Row';
import './../App.css';

export default class Table extends React.Component {
	constructor(props) {
	  super(props);
	  this.state = {
			cols: null,
			rows: null,
			showTable : false,
			empName : '',
			showError : false
	  };	  

	}
	
	compareBy = (key) => {
	  return function (a, b) {
		if (a[key] < b[key]) return -1;
		if (a[key] > b[key]) return 1;
		return 0;
	  };
	}
   
	sortBy = (key) => {
	  let arrayCopy = [...this.state.rows];
	  arrayCopy.sort(this.compareBy(key));
	  this.setState({rows: arrayCopy});
	}

	fileHandler = (event) => {
		let fileObj = event.target.files[0];
			if(fileObj){
				//just pass the fileObj as parameter
				ExcelRenderer(fileObj, (err, resp) => {  
					resp.rows  = resp.rows.map(	
						(item) =>
						{
							return {
								'sr_no' : item[0],
								'title' : item[1],
								'website' : item[2],
								'result' : item[3],
								'gen' : item[4],
								'obc' : item[5]
							};	  
						}				  
					);
					
					if(err){
						console.log(err);            
					}
					else{
						this.setState({
						cols: resp.cols,
						rows: resp.rows,
						showTable : true
						});
					}
				});
		}		
	}

	getEmpData = () => {
		console.log(this.empNameRef.value);
		var temp_Emp = [...this.state.rows];
		if(this.empNameRef.value.length < 1){
			this.setState({
				showError : true
			});
		}else{
			temp_Emp = temp_Emp.filter( emp => emp.title.toLowerCase() === this.empNameRef.value.toLowerCase());
			this.setState({
				empName : temp_Emp[0]
			});
		}
			
	}
	  
	render() {
		console.log('Rendered');
	  const rows = this.state.rows && this.state.rows.map(
		  (rowData) =>
			<Row {...rowData} />	  		
	  );
  
	  return (
		  <div> Please upload details here : 
			<input type="file" onChange={this.fileHandler} style={{"padding":"10px"}} />			
			{this.state.showTable && <div className="table">
				<div className="header">
					<div id="empId" onClick={() => this.sortBy('sr_no')} >empId</div>
					<div id="title" onClick={() => this.sortBy('title')}>empName</div>
					<div id="website" onClick={() => this.sortBy('website')}>salary in $</div>
					<div id="result" onClick={() => this.sortBy('result')}>Age</div>
					<div id="gen" onClick={() => this.sortBy('gen')}>department</div>
					<div id="obc" onClick={() => this.sortBy('obc')}>title</div>
				</div>
				<div className="body">
					{rows}
				</div>
				<div>
				
				<br></br><br></br>	Search Employee : <br></br>{this.state.showError && <span>Please Enter username</span>}
				<br></br><input type="text" name="empname" ref={ (input) => this.empNameRef = input}/>
					
					<input type="submit" onClick={this.getEmpData} value="Search"/>
					<br>
					</br><br></br>
					<div>User Details : {JSON.stringify(this.state.empName)}</div>
						
				</div>
			</div>
			}
		  </div>
		
	  );
	  
	}
}