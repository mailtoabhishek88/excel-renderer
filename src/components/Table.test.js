import React from 'react';
import { shallow} from 'enzyme';
import Table from './Table';

describe('Table Component', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Table debug/>);
    console.log('---------',Object.keys(component));
    expect(component).toMatchSnapshot();
  });
});