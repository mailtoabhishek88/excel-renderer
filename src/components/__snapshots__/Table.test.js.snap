// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Table
    debug={true}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        " Please upload details here :",
        <input
          onChange={[Function]}
          style={
                    Object {
                              "padding": "10px",
                            }
          }
          type="file"
/>,
        false,
      ],
    },
    "ref": null,
    "rendered": Array [
      " Please upload details here :",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "style": Object {
            "padding": "10px",
          },
          "type": "file",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      false,
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          " Please upload details here :",
          <input
            onChange={[Function]}
            style={
                        Object {
                                    "padding": "10px",
                                  }
            }
            type="file"
/>,
          false,
        ],
      },
      "ref": null,
      "rendered": Array [
        " Please upload details here :",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": [Function],
            "style": Object {
              "padding": "10px",
            },
            "type": "file",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        false,
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": false,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
